theme/static/js/hs_file_browser.js

```diff
@@ -919,14 +919,28 @@ $(document).ready(function () {
         });
     });
 
-     // Download method
+    // Download method
     $("#btn-download, #fb-download").click(function () {
         var downloadList = $("#fb-files-container li.ui-selected");
+
+        // Remove previous temporary download frames
+        $(".temp-download-frame").remove();
+
         if (downloadList.length) {
-            for (var i = 0; i < downloadList.length; i++) {
-                var url = $(downloadList[i]).attr("data-url");
-                var fileName = $(downloadList[i]).children(".fb-file-name").text();
-                downloadURI(url, fileName);
+            // Workaround for Firefox and IE
+            if (jQuery.browser.mozilla == true || !!navigator.userAgent.match(/Trident\/7\./)) {
+                for (var i = 0; i < downloadList.length; i++) {
+                    var url = $(downloadList[i]).attr("data-url");
+                    var frameID = "download-frame-" + i;
+                    $("body").append("<iframe class='temp-download-frame' id='" + frameID + "' style='display:none;' src='" + url + "'></iframe>");
+                }
+            }
+            else {
+                for (var i = 0; i < downloadList.length; i++) {
+                    var url = $(downloadList[i]).attr("data-url");
+                    var fileName = $(downloadList[i]).children(".fb-file-name").text();
+                    downloadURI(url, fileName);
+                }
```

Comment:
While that would be less code, it would be inefficient since the conditional to check for the browser would be checked at every iteration. They way I have it the check is done only once and the corresponding for loop runs. So I think the way I have it is correct in this case.